c/driver/postgresql/connection.cc

```diff
@@ -145,6 +145,40 @@ AdbcStatusCode PostgresConnection::GetInfo(struct AdbcConnection* connection,
   return BatchToArrayStream(&array, &schema, out, error);
 }
 
+AdbcStatusCode PostgresConnectionGetSchemasImpl(PGconn* conn, int depth,
+                                                struct ArrowArray* db_schemas_col,
+                                                struct AdbcError* error) {
+  struct ArrowArray* db_schema_items = db_schemas_col->children[0];
+  struct ArrowArray* schema_name_col = db_schema_items->children[0];
+  struct ArrowArray* schema_tables_col = db_schema_items->children[1];
+
+  const char* stmt =
+      "SELECT nspname FROM pg_catalog.pg_namespace WHERE "
+      "nspname !~ '^pg_' AND nspname <> 'information_schema'";
+  auto result_helper = PqResultHelper(conn, stmt);
+
+  pg_result* result = result_helper.Execute();
+  ExecStatusType pq_status = PQresultStatus(result);
+  if (pq_status == PGRES_TUPLES_OK) {
```

Comment:
This pg looping is getting pretty repetitive. Feels like there is a way to refactor into the existing ResultHelper class and create an iterator over the results